code/modules/karma/karma_holder.dm

```diff
@@ -0,0 +1,423 @@
+// This file contains stuff regarding the karma holder datum
+// This is attached to a client and holds their info
+/datum/karma_holder
+	/// List of jobs this person has unlocked
+	var/list/unlocked_jobs = list()
+	/// List of species this person has unlocked
+	var/list/unlocked_species = list()
+	/// Amount of karma earned total
+	var/karma_earned = 0
+	/// Amount of karma spent total
+	var/karma_spent = 0
+	/// What tab are they on
+	var/karma_tab = 0
+
+
+// Safety stuff
+/datum/karma_holder/vv_edit_var(var_name, var_value)
+	return FALSE // no
+
+/datum/karma_holder/CanProcCall(procname)
+	return FALSE // no
+
+// Opens the karma shop menu
+/datum/karma_holder/proc/open_shop_menu()
+	var/dat = "<html><body><center>"
+	dat += "<a href='?src=[UID()];karmashop=tab;tab=0' [karma_tab == 0 ? "class='linkOn'" : ""]>Job Unlocks</a>"
+	dat += "<a href='?src=[UID()];karmashop=tab;tab=1' [karma_tab == 1 ? "class='linkOn'" : ""]>Species Unlocks</a>"
+	dat += "<a href='?src=[UID()];karmashop=tab;tab=2' [karma_tab == 2 ? "class='linkOn'" : ""]>Karma Refunds</a>"
+	dat += "</center>"
+	dat += "<HR>"
+
+	var/currentkarma = (karma_earned - karma_spent)
+	dat += "You have <b>[currentkarma]</b> available.<br><HR>"
+
+	switch(karma_tab)
+		if(0) // Job Unlocks
+			if(!("Barber" in unlocked_jobs))
```

Comment:
I hate that all this is hardcoded.

Reply:
Same. It might not be for much longer though.